File: ApemanPlugin/src/main/kotlin/apeman_core/features_extraction/calculators/candidate/NumIfCandidateCalculator.java
package apeman_core.features_extraction.calculators.candidate;

import apeman_core.base_entities.FeatureType;
import apeman_core.pipes.CandidateWithFeatures;
import com.intellij.psi.JavaRecursiveElementVisitor;
import com.intellij.psi.PsiIfStatement;

import java.util.ArrayList;

public class NumIfCandidateCalculator extends AbstractNumCandidateCalculator {

    public NumIfCandidateCalculator(ArrayList<CandidateWithFeatures> candidates) {
        super(candidates, FeatureType.NUM_IF);
    }

    @Override
    public JavaRecursiveElementVisitor createVisitor() {
        return new NumIfCandidateCalculator.Visitor();
    }

    private class Visitor extends CandidateVisitor {

        @Override
        public void visitIfStatement(PsiIfStatement statement) {
            super.visitIfStatement(statement);
            if (isInsideMethod) {
                updateCounters();
            }
        }
    }
}

File: ApemanPlugin/src/main/kotlin/apeman_core/features_extraction/calculators/candidate/NumIfCandidateCalculator.kt
package apeman_core.features_extraction.calculators.candidate

import apeman_core.base_entities.ExtractionCandidate
import apeman_core.base_entities.FeatureType
import com.intellij.psi.PsiIfStatement

class NumIfCandidateCalculator(candidates: List<ExtractionCandidate>
) : AbstractNumCandidateCalculator(candidates, FeatureType.NUM_IF) {

    override fun createVisitor(methodCandidates: List<ExtractionCandidate>) = Visitor(methodCandidates)

    inner class Visitor(methodCandidates: List<ExtractionCandidate>
    ) : AbstractNumCandidateCalculator.CandidateVisitor(methodCandidates) {

        override fun visitIfStatement(statement: PsiIfStatement) {
            super.visitIfStatement(statement)
            if (isInsideMethod) {
                updateCounters()
            }
        }
    }
}
